docs/configuration.md

```diff
@@ -78,10 +74,36 @@ spec:
   template:
     spec:
       securityGroups:
-      - name: k8s-cluster-${NAMESPACE}-${CLUSTER_NAME}-secgroup-controlplane
       - name: allow-ssh
 ```
 
+## OpenStack credential
+
+The [env.rc](./env.rc) script sets the environment variables related to credentials.
+
+```bash
+source env.rc <path/to/clouds.yaml> <cloud>
+```
+
+The following variables are set.
+
+| Variable | Meaning |
+ :----- | :-------- 
+| OPENSTACK_CLOUD | The cloud name which is used as second argument |
+| OPENSTACK_CLOUD_YAML_B64 | The secret used by Cluster API Provider OpenStack accessing OpenStack |
+| OPENSTACK_CLOUD_PROVIDER_CONF_B64 | The content of [cloud.conf](https://kubernetes.io/docs/concepts/cluster-administration/cloud-providers/#cloud-conf) which is used by OpenStack cloud provider |
+| OPENSTACK_CLOUD_CACERT_B64 | (Optional) The content of your custom CA file which can be specified in your clouds.yaml by `ca-file` |
+
+## Avaialability zone
+
+The availability zone names must be exposed as an environment variable `OPENSTACK_FAILURE_DOMAIN`.
+
+## DNS server
+
+The DNS servers must be exposed as an environment variable `OPENSTACK_DNS_NAMESERVERS`.
```

Comment:
is it true? must expose?

Reply:
Yes, this is true!
```bash
(python) ubuntu@k8s:~/cluster-api-provider-openstack$ ./bin/clusterctl config cluster capi-quickstart --kubernetes-version v1.17.3 --control-plane-machine-count=3 --worker-machine-count=3 --infrastructure openstack:v0.3.0 > test.yaml
Error: failed to perform variable substitution: value for variables [OPENSTACK_DNS_NAMESERVERS] is not set. Please set the value using os environment variables or the clusterctl config file
```

Reply:
oops... I mean, technically , not whether we are checking it 
e.g whether the DNS is really needed.. not remember clearly

Reply:
we can create subnet without DNS server. But kubeadm needs dns resolution. The VM image have to be created with DNS settings if `OPENSTACK_DNS_NAMESERVERS` is not given.

Reply:
ok, works for me , thanks~